src/appleseed-max-impl/builtinmapsupport.cpp

```diff
@@ -43,6 +44,47 @@
 namespace asf = foundation;
 namespace asr = renderer;
 
+void connect_output_selector(
+    renderer::ShaderGroup&  shader_group,
+    const char*             material_node_name,
+    const char*             material_input_name,
+    Texmap*                 texmap,
+    const Color             const_value)
+{
+    const auto t = GetCOREInterface()->GetTime();
+
+    Texmap* input_map = nullptr;
+    get_paramblock_value_by_name(texmap->GetParamBlock(0), L"source_map", t, input_map, FOREVER);
+    int output_index = 0;
+    get_paramblock_value_by_name(texmap->GetParamBlock(0), L"output_index", t, output_index, FOREVER);
+
+    if (input_map != nullptr && is_osl_texture(input_map))
+    {
+        auto output_names = static_cast<OSLTexture*>(input_map)->get_output_names();
+
+        static_cast<OSLTexture*>(input_map)->create_osl_texture(
+            shader_group,
+            material_node_name,
+            material_input_name,
+            static_cast<int>((output_index - 1) % output_names.size()));
+    }
+}
+
+void connect_output_selector(
+    renderer::ShaderGroup&  shader_group,
+    const char*             material_node_name,
+    const char*             material_input_name,
+    Texmap*                 texmap,
+    const float             const_value)
+{
+    connect_output_selector(
+        shader_group,
+        material_node_name,
+        material_input_name,
+        texmap,
+        Color());
```

Comment:
Is this correct? Shouldn't it be `Color(const_value)`? Just checking...

Reply:
Last parameter is discarded in both version so override is not needed here.. Thanks for pointing that out!